Gui/opensim/view/src/org/opensim/threejs/ModelVisualizationJson.java

```diff
@@ -920,7 +921,8 @@ public JSONObject createAppearanceMessage(Component mc, AbstractProperty prop) {
             JSONObject commandJson = CommandComposerThreejs.createAppearanceChangeJson(prop, objectUuid);
             guiJson.put("command", commandJson);
             String commandName = (String) commandJson.get("name");
-            if (Muscle.safeDownCast(mc)!=null && commandName.equalsIgnoreCase("SetVisible"))
+            Force mcAsForce = Force.safeDownCast(mc);
+            if (mcAsForce!=null && mcAsForce.hasGeometryPath() && commandName.equalsIgnoreCase("SetVisible"))
                commandJson.put("type", "SetValueCommandMuscle");
```

Comment:
It seems like the visualizer knows only of muscles (based on the string "SetValueCommandMuscle"). It seems that the visualizer should handle paths, not muscles. Is this simply a matter of naming?

Reply:
Exactly it's naming, muscles mean nothing more than a path to the visualizer.

Reply:
Can you create an issue for renaming/refactoring to improve the clarity of the code?

Reply:
The JS object on the visualizer side though is called SkinnedMuscle so the language on the visualizer side is consistent, but feel free to open an issue regarding clarity of this or any other piece of code you review on the gui side. Thank you @chrisdembia